lib/std/special/compiler_rt/README.md

```diff
@@ -1,15 +1,258 @@
+- __fixdfti      //
+- __fixtfdi      //
+- __fixxfti      // missing
+
+- __fixunssfsi   // convert to u32, rounding towards zero. negative values become 0.
+- __fixunsdfsi   //
+- __fixunstfsi   //
+- __fixunsxfsi   // missing
+- __fixunssfdi   // convert to u64, rounding towards zero. negative values become 0.
+- __fixunsdfdi   //
+- __fixunstfdi   //
+- __fixunsxfdi   // missing
+- __fixunssfti   // convert to u128, rounding towards zero. negative values become 0.
+- __fixunsdfti   //
+- __fixunstfdi   //
+- __fixunsxfti   // missing
+
+- __floatsisf    // convert i32 to floating point
+- __floatsidf    //
+- __floatsitf    //
+- __floatsixf    // missing
+- __floatdisf    // convert i64 to floating point
+- __floatdidf    //
+- __floatditf    //
+- __floatdixf    // missing
+- __floattisf    // convert i128 to floating point
+- __floattidf    //
+- __floattixf    //
+
+- __floatunsisf  // convert i32 to floating point
+- __floatunsidf  //
+- __floatunsitf  //
+- __floatunsixf  // missing
+- __floatundisf  // convert i64 to floating point
+- __floatundidf  //
+- __floatunditf  //
+- __floatundixf  // missing
+- __floatuntisf  // convert i128 to floating point
+- __floatuntidf  //
+- __floatuntitf  //
+- __floatuntixf  // missing
+
+#### Comparison
+- __cmpsf2       // return (a<b)=>-1,(a==b)=>0,(a>b)=>1,Nan=>1 dont rely on this
+- __cmpdf2       // exported from __lesf2, __ledf2, __letf2 (below)
+- __cmptf2       //
+- __unordsf2     // (input==NaN) => out!=0 else out=0,
+- __unorddf2     // __only reliable for (input!=Nan)__
+- __unordtf2     //
+- __eqsf2        // (a!=NaN) and (b!=Nan) and (a==b) => output=0
+- __eqdf2        //
+- __eqtf2        //
+- __nesf2        // (a==NaN) or (b==Nan) or (a!=b) => output!=0
+- __nedf2        //
+- __netf2        //
+- __gesf2        // (a!=Nan) and (b!=Nan) and (a>=b) => output>=0
+- __gedf2        //
+- __getf2        //
+- __ltsf2        // (a!=Nan) and (b!=Nan) and (a<b) => output<0
+- __ltdf2        //
+- __lttf2        //
+- __lesf2        // (a!=Nan) and (b!=Nan) and (a<=b) => output<=0
+- __ledf2        //
+- __letf2        //
+- __gtsf2        // (a!=Nan) and (b!=Nan) and (a>b) => output>0
+- __gtdf2        //
+- __gttf2        //
+
+#### Arithmetic
+- __addsf3       // a + b f32
+- __adddf3       // a + b f64
+- __addtf3       // a + b f128
+- __addxf3       // a + b f80
+- __aeabi_fadd   // a + b f64 ARM: AAPCS
+- __aeabi_dadd   // a + b f64 ARM: AAPCS
+- __subsf3       // a - b
+- __subdf3       // a - b
+- __subtf3       // a - b
+- __subxf3       // a - b f80
+- __aeabi_fsub   // a - b f64 ARM: AAPCS
+- __aeabi_dsub   // a - b f64 ARM: AAPCS
+- __mulsf3       // a * b
+- __muldf3       // a * b
+- __multf3       // a * b
+- __mulxf3       // a * b missing
+- __divsf3       // a / b
+- __divdf3       // a / b
+- __divtf3       // a / b
+- __divxf3       // a / b missing
+- __negsf2       // -a
+- __negdf2       // -a
+- __negtf2       // -a missing
+- __negxf2       // -a missing
```

Comment:
These don't exist because calls to them are never generated. LLVM lowers `__negtf2` directly into an xor and we do the same for `f80` in `ir_gen_soft_f80_neg` which regressed with #10861.

Here's a patch to fix it:
```diff
diff --git a/src/stage1/codegen.cpp b/src/stage1/codegen.cpp
index dc8f5e590..8bb5430ae 100644
--- a/src/stage1/codegen.cpp
+++ b/src/stage1/codegen.cpp
@@ -4115,13 +4115,9 @@ static LLVMValueRef ir_render_binary_not(CodeGen *g, Stage1Air *executable,
 static LLVMValueRef ir_gen_soft_f80_neg(CodeGen *g, ZigType *op_type, LLVMValueRef operand) {
     uint32_t vector_len = op_type->id == ZigTypeIdVector ? op_type->data.vector.len : 0;
 
-    uint64_t buf[2] = {0, 0};
-    if (g->is_big_endian != native_is_big_endian) {
-        buf[1] = 0x8000000000000000;
-    } else {
-        buf[1] = 0x8000;
-    }
-    LLVMValueRef sign_mask = LLVMConstIntOfArbitraryPrecision(LLVMInt128Type(), 2, buf);
+    LLVMTypeRef llvm_i80 = LLVMIntType(80);
+    LLVMValueRef sign_mask = LLVMConstInt(llvm_i80, 1, false);
+    sign_mask = LLVMConstShl(sign_mask, LLVMConstInt(llvm_i80, 79, false));
 
     LLVMValueRef result;
     if (vector_len == 0) {
```

Reply:
I can make a PR, if I know what to test for. Do you have the test cases in arocc?

Reply:
I added the parts of negtf2 and negxf2 being unnecessary with your written reason.

Reply:
@Vexu applied the patch in 6235afc6, so this is fixed.